src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,13 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            private fixed char _cFileName[260];
+            private fixed char _cAlternateFileName[14];
+
+            internal Span<char> cFileName
+            {
+                get { fixed (char* c = _cFileName) return new Span<char>(c, 260); }
+            }
```

Comment:
>Note that this Span<char>-based approach needs to be used with care

One answer to this is to encapsulate the functionality in a struct, which is one of the key reasons I went with that sort of design to begin with. I get the argument of trying to utilize the work we've done / are doing with Span. I _could_ redo the original concept and utilize span internally to get fully safe and nearly as performant as I started with. Given how suboptimal our P/Invokes are and how easy it is to write them incorrectly I still strongly prefer bullet-proof- which we can get with encapsulation. I will, however, defer to whatever our majority opinion is so that I can confidently move forward on fixing our I/O perf.

I will concede, however, that using the span out of scope is likely to fail hard and fast, so that part isn't that terrifying. Exposing as a Span does, however, make it easy to use these fixed buffers incorrectly. Null termination is easy to slip up on. The buffer may not have a null at end, for one. Another slip-up example is that "foo\0bar" does **not** equal a buffer of 'f' 'o' 'o' '\0' 'b' 'a' 'r' (which is actually "foo").

@jkotas, what do you prefer? Are you still happiest with exposing a Span accessor given the above discussion?